db/commitlog/commitlog.cc

```diff
@@ -442,6 +484,11 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     seastar::gate _gate;
     uint64_t _new_counter = 0;
     std::optional<size_t> _disk_write_alignment;
+    std::deque<message> _message_queue;
```

Comment:
What limits this from growing without bounds?

Reply:
Nothing. But using a deque to at least partially deal with potentially large message buffer. 
In practice, we don't see a large message queue, since actual execution of messages are not delayed, thus the real overflow risk is probably in the pending future queue (built implicitly).